### PR TITLE
Hotfix for Release 2.0.4

### PR DESCRIPTION
**Pull request by Antiglobalist**
Discussion - https://github.com/SceneView/sceneview-android/issues/435
Issues:
- lifecycle?.coroutineScope can be isActive == false and destroy() will not be called 
- destroyARCore was calling twice. It produced NDK crash from ARCore

The second call of destroy() was removed to prevent crash
Coroutines were replaced to ExecutorService

### CODE REVIEW

File: arsceneview/src/main/java/io/github/sceneview/ar/ARSceneView.kt [hunk 1]
```diff
@@ -565,8 +562,6 @@ open class ARSceneView @JvmOverloads constructor(
 
     override fun destroy() {
         if (!isDestroyed) {
-            destroyARCore()
```
Comment (Antiglobalist) [inline arsceneview/src/main/java/io/github/sceneview/ar/ARSceneView.kt:568]: You use LifeCycleObserver as the main place to call ArCore lifecycle functions
And you don't need to call it here again
I didn't find any side cases

File: arsceneview/src/main/java/io/github/sceneview/ar/ARSceneView.kt [hunk 2]
```diff
@@ -577,14 +572,6 @@ open class ARSceneView @JvmOverloads constructor(
         super.destroy()
     }
 
-    fun destroyARCore() {
-        val scope = lifecycle?.coroutineScope ?: CoroutineScope(Dispatchers.IO)
```
Comment (Antiglobalist) [inline arsceneview/src/main/java/io/github/sceneview/ar/ARSceneView.kt:581]: As I wrote in issue - Coroutines is not fit tool to execute background operated a specialty on Destroy callback

File: arsceneview/src/main/java/io/github/sceneview/ar/ARSceneView.kt [hunk 3]
```diff
@@ -577,14 +572,6 @@ open class ARSceneView @JvmOverloads constructor(
         super.destroy()
     }
 
-    fun destroyARCore() {
```
Comment (Antiglobalist) [inline arsceneview/src/main/java/io/github/sceneview/ar/ARSceneView.kt:580]: No reasons to place public function cuz it's internal behavior 
Same as all other lifecycle calls

**Review by Antiglobalist — COMMENTED** [review]
Notes

**Comment by sameerjj** [comment]
looks good to me

**Review by ThomasGorisse — APPROVED** [review]

**Pull request merged**